### PR TITLE
Bug254

### LINKED ISSUE #254: yaml round trip error
Getting this error from an msapp:
`Error   PA3003: Parse error: Property should be at same indent level at Workflows.fx.yaml:2285,17-2285,17`

### PR DESCRIPTION
**Pull request by MikeStall**
Fix #254 
Round trip \r\n pattern.

### CODE REVIEW

File: src/PAModel/PAConvert/Yaml/YamlWriter.cs [hunk 1]
```diff
@@ -110,6 +110,13 @@ public void WriteProperty(string propertyName, string value)
                         WriteIndent();
                         needIndent = false;
                     }
+
+                    if (ch == '\r')
```
Comment (lesaltzm) [inline src/PAModel/PAConvert/Yaml/YamlWriter.cs:114]: \r shouldn't be making it to the yamlwriter, see a couple places in IRStateHelper where we do something like  
`           var script = rule.InvariantScript.Replace("\r\n", "\n").Replace("\r", "\n").TrimStart();`

What kind of property is hitting this and why is `\r\n` making its way into the IR to be written?

Comment (lesaltzm) [reply src/PAModel/PAConvert/Yaml/YamlWriter.cs:114]: Maybe the right thing to do here is to define an actual constructor for ExpressionNode and do this trimming/normalizing there.

Comment (MikeStall) [reply src/PAModel/PAConvert/Yaml/YamlWriter.cs:114]: The yaml layer should handle \r  (we can't rely on the upstream caller to remove them).  Ie, see new test added in this pr. 
In this bug, there's probably some path that's skipping the IR's replacement.

Comment (MikeStall) [reply src/PAModel/PAConvert/Yaml/YamlWriter.cs:114]: We're definitely missing a bunch of spots in the IRStateHelper. 
Although if we do it in the lexer, then we should be able to completely remove the conversions from IRStateHelper . I'll try that ...

**Review by lesaltzm — APPROVED** [review]

**Pull request merged**